packages/block/src/types.ts

```diff
@@ -35,6 +36,13 @@ export interface BlockOptions {
    * Default: `false`
    */
   initWithGenesisHeader?: boolean
+
+  /**
+   * If a preceding `BlockHeader` (usually the parent header) is given the preceding
+   * header will be used to calculate the difficulty for this block and the calculated
+   * difficulty takes precedence over a provided static `difficulty` value.
+   */
+  calcDifficultyFromHeader?: BlockHeader
```

Comment:
>If a `BlockHeader` is given, then this header is used to calculate the difficulty upon creating the Block Header.

This was to prone yet for misunderstanding, have updated the comment with an additional commit. The double `Block Header` reference was too confusing - one refering to the header provided and one to the header created.

Some general note: I think the usage of "this" in these kind of option comments often don't work, since it is always a bit misleading in the sense that it becomes not 100% clear if "this" refers to "this" option provided or to "this" instance created, I now stumbled upon this (haha 😄 ) a couple of times. Think it works better to in doubt be a bit more redundant and e.g. repeat "the header provided" instead of using "this header".

Reply:
Ah yes I'm somewhat used to write `this` in these descriptions, will have to self-review those sometimes 👍